File: 485.max-consecutive-ones.java
/*
 * @lc app=leetcode id=485 lang=java
 *
 * [485] Max Consecutive Ones
 */

// @lc code=start
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxCount = 0; // Biến để lưu trữ số lượng 1 liên tiếp lớn nhất
        int currentCount = 0; // Biến để đếm số lượng 1 liên tiếp hiện tại
        for (int num : nums) { // Duyệt qua từng phần tử trong mảng
            if (num == 1) { // Nếu phần tử là 1
                currentCount++; // Tăng số lượng 1 liên tiếp hiện tại
            } else { // Nếu phần tử không phải là 1
                maxCount = Math.max(maxCount, currentCount); // Cập nhật số lượng 1 liên tiếp lớn nhất nếu cần
                currentCount = 0; // Đặt lại số lượng 1 liên tiếp hiện tại về 0
            }
        }
        return Math.max(maxCount, currentCount); // Trả về số lượng 1 liên tiếp lớn nhất
    } 
}
// @lc code=end

File: 485.max-consecutive-ones.py
#
# @lc app=leetcode id=485 lang=python3
#
# [485] Max Consecutive Ones
#

# @lc code=start
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        if not nums: # Kiểm tra danh sách rỗng
            return 0 # Trả về 0 nếu danh sách rỗng
        max_count = 0 # Biến lưu trữ số 1 liên tiếp lớn nhất
        current_count = 0 # Biến lưu trữ số 1 liên tiếp hiện tại
        for num in nums: # Duyệt qua từng phần tử trong danh sách
            if num == 1: # Nếu phần tử là 1
                current_count += 1 # Tăng số 1 liên tiếp hiện tại
                max_count = max(max_count, current_count) # Cập nhật số 1 liên tiếp lớn nhất
            else: # Nếu phần tử không phải là 1
                current_count = 0 # Đặt lại số 1 liên tiếp hiện tại
        return max_count # Trả về số 1 liên tiếp lớn nhất
# @lc code=end

